identifyyeastcells.py

```diff
@@ -0,0 +1,1146 @@
+
+        ui_params.append(params["segmentation"]["stars"]["points"])
+        ui_params.append(1.0 / params["segmentation"]["stars"]["step"])
+        ui_params.append(len(params["segmentation"]["stars"]["sizeWeight"]))
+
+        return ui_params
+
+    def set_ui_from_precision(self, ui_precision):
+        ui_precision_values = self.get_ui_params_from_precision(ui_precision)
+        ui_precision_settings = self.settings()[self.PRECISION_PARAMS_START:self.PRECISION_PARAMS_END+1]
+        for setting, value in zip(ui_precision_settings, ui_precision_values):
+            setting.value = value
+
+
+    def prepare_cell_star_object(self, segmentation_precision):
+        cellstar = Segmentation(segmentation_precision, self.average_cell_diameter.value)
+        cellstar.parameters["segmentation"]["maxOverlap"] = self.maximal_cell_overlap.value
+        self.set_params_from_ui(cellstar.parameters)
+
+        if self.advanced_cell_filtering.value:
+            def calculate_area_multiplier(area):
+                return 4.0 * area / self.average_cell_diameter.value ** 2 / math.pi
+
+            #def calculate_size_multiplier(area):
+            #    return calculate_area_multiplier(area) ** 0.5
+
+            areas_range = self.min_cell_area.value, self.max_cell_area.value
+            cellstar.parameters["segmentation"]["minArea"] = max(cellstar.parameters["segmentation"]["minArea"], calculate_area_multiplier(areas_range[0]))
+            cellstar.parameters["segmentation"]["maxArea"] = calculate_area_multiplier(areas_range[1])
+            # to some extent change length of rays
+            #cellstar.parameters["segmentation"]["stars"]["maxSize"] = max(cellstar.parameters["segmentation"]["stars"]["maxSize"], min(2.5, calculate_size_multiplier(areas_range[1])))
+
+        success = cellstar.decode_auto_params(self.autoadapted_params.value)
+        if not success:  # if current value is invalid overwrite it with current settings
+            self.autoadapted_params.value = cellstar.encode_auto_params()
+        return cellstar
+
+    def preprocess_images(self, input_image, background_image, ignore_mask):
+        # Invert images if required.
+        if not self.background_brighter_then_cell_inside:
+            input_image = 1 - input_image
+            if background_image is not None:
+                background_image = 1 - background_image
+
+        # support for fluorescent images
+        # here it is design question: we assume that the user *should* say
+        # truth about background and inside of cells: insides are brighter
+        # than background (so the bkg and image for fluorescent will be
+        # inverted at this stage)
+        if not self.bright_field_image:
+            sigma = 4
+            size = int(sigma * 4) + 1
+            mask = np.ones(input_image.shape, bool)
+            edge_pixels = laplacian_of_gaussian(input_image, mask, size, sigma)
```

Comment:
Part of the upcoming work we're planning includes removing anything in centrosome that's available in numpy/scipy/scikit-image. This will help align the project with these core image processing and scientific computing libraries.